content/_guides/sasl/emacs-erc.md

```diff
@@ -0,0 +1,156 @@
+```
+;; ======== ERC START ==============================
+
+;; Most of these are just for convenience options. I believe the only
+;; absolutely necessary one here is the my-erc-connect() function I
+;; define for connecting.
+
+;; for auto-joining networks and channels when starting up ERC
+(require 'erc-join)
+(erc-autojoin-enable)
+
+;; Logging of conversations (mostly for when I forget to save peoples’
+;; answers when I ask a programming question).
+
+;; turn on logging
+(setq erc-log-mode t)
+;; write to log file when new text is added to a logged ERC buffer.
+(setq erc-log-write-after-insert t)
+;; write to log file after every message I send.
+(setq erc-log-write-after-send t)
+;; Include network name in the buffers
+(setq erc-rename-buffers t)
+
+;; from local erc-log.el and tweeked
+;; see also https://www.emacswiki.org/emacs/ErcLogging
+(require 'erc-log)
+(files--ensure-directory "~/.config")
+(files--ensure-directory "~/.config/erc")
+(files--ensure-directory "~/.config/erc/logs")
+(setq erc-log-channels-directory "~/.config/erc/logs/")
+(erc-log-enable)
+
+;; don't ask me for my nickserv password
+(setq erc-prompt-for-nickserv-password nil)
+
+;; A list of channels to auto-join
+(setq erc-autojoin-channels-alist
+	'(("irc.libera.chat" "#emacs" "#guile" "#scheme" "#erc" "#godot")))
+;; OR, if NONE..
+;; (setq erc-autojoin-channels-alist nil)
+
+;; from http://www.john2x.com/emacs.html
+;; Set the prompt to use the channel name.
+(setq erc-prompt  (lambda () (concat (buffer-name) "> ")))
+
+;; beautify the text on screen
+(setq erc-fill-function 'erc-fill-static)
+(setq erc-fill-static-center 10)
+
+;; my main function for connecting to libera.chat
+;; (see also https://libera.chat/guides/connect for regional hostnames)
+(defun my-erc-connect ()
+  (interactive)
+  (erc-tls :server "irc.libera.chat" :port 6697 :nick "MYNICK" :full-name "MY NAME"))
+(global-set-key (kbd "C-c e") 'my-erc-connect)
+
+;; Channel-tracking: Convenience functions for easily flipping between channels you are logged into
+;; from https://www.emacswiki.org/emacs/ErcChannelTracking#toc8
+;; (I don't use so much anymore, actually..)
+(defvar erc-channels-to-visit nil
+    "Channels that have not yet been visited by erc-next-channel-buffer")
+(defun erc-next-channel-buffer ()
+  "Switch to the next unvisited channel. See erc-channels-to-visit"
+  (interactive)
+  (when (null erc-channels-to-visit)
+    (setq erc-channels-to-visit 
+          (remove (current-buffer) (erc-channel-list nil))))
+  (let ((target (pop erc-channels-to-visit)))
+    (if target 
+        (switch-to-buffer target))))
+(global-set-key (kbd "C-c n") 'erc-next-channel-buffer)
+;; ======== ERC END ================================
+```
+3. In the my-erc-connect command you've defined, replace MYNICK with your nickname you plan to use (assuming no one else uses and has registered this nick).
```

Comment:
leave a blank row between the end of the code fence and the next row
````suggestion
```

3. In the my-erc-connect command you've defined, replace MYNICK with your nickname you plan to use (assuming no one else uses and has registered this nick).
````

Reply:
okay, done